Source/JavaScriptCore/runtime/JSArray.cpp

```diff
@@ -1396,7 +1400,13 @@ JSArray* constructArrayNegativeIndexed(JSGlobalObject* globalObject, Structure*
     VM& vm = globalObject->vm();
     ObjectInitializationScope scope(vm);
 
-    JSArray* array = constructArray(scope, arrayStructure, length);
+    JSArray* array = constructArray<AllocationFailureMode::ReturnNull>(scope, arrayStructure, length);
+    if (UNLIKELY(!array)) {
+        auto scope = DECLARE_THROW_SCOPE(vm);
```

Comment:
The ThrowScope should be declared at the top right after the declaration of `VM& vm`.  The goal of the ThrowScope is to help us detect clients of this function that needs to handle the potential exception.  Declaring it inside an if statement defeats this.

Please also run the jsc stress tests on a Debug build locally that will do exception verification checks.  I think the EWS runs a Release build.

Reply:
Thank you for review Mark, I've hoisted `ThrowScope` declaration and fixed a client to handle potential exception.

> Please also run the jsc stress tests on a Debug build locally that will do exception verification checks. I think the EWS runs a Release build.

Done for `--filter "arr"`.